MapboxMobileEvents/Categories/NSProcessInfo+SystemInfo.m

```diff
@@ -0,0 +1,65 @@
+#import "NSProcessInfo+SystemInfo.h"
+#include <sys/types.h>
+#include <sys/sysctl.h>
+#include <mach/machine.h>
+
+@implementation NSProcessInfo (SystemInfo)
+
++ (NSString*)stringForProcessorType:(NSInteger)type subtype:(NSInteger)subtype {
+
+    // Scratch String to build description. Ensures non-nil
+    NSMutableString *cpu = [[NSMutableString alloc] init];
+
+    // values for cputype and cpusubtype defined in mach/machine.h
+    if (type == CPU_TYPE_X86_64) {
+        [cpu appendString:@"x86_64"];
+    } else if (type == CPU_TYPE_X86) {
+        [cpu appendString:@"x86"];
+    } else if (type == CPU_TYPE_ARM) {
+        [cpu appendString:@"ARM"];
+        switch(subtype)
+        {
+            case CPU_SUBTYPE_ARM_V6:
+                [cpu appendString:@"V6"];
```

Comment:
Can we make these `v6` etc so it's `ARMv6` like in their docs: https://static.docs.arm.com/ddi0419/d/DDI0419D_armv6m_arm.pdf

Reply:
Sure thing.

Did you have any thoughts on the subtype E? I wasn't sure if that meant bionic? Cortext?

Reply:
I was just comparing with our supported architectures:

`archs:           [ armv7, armv7s, arm64, arm64e ]`

How would you feel about an all lowercased output?

Reply:
eek, looks like I missed a few more sub type variants:

```
#define CPU_SUBTYPE_ARM_ALL             ((cpu_subtype_t) 0)
#define CPU_SUBTYPE_ARM_V4T             ((cpu_subtype_t) 5)
#define CPU_SUBTYPE_ARM_V6              ((cpu_subtype_t) 6)
#define CPU_SUBTYPE_ARM_V5TEJ           ((cpu_subtype_t) 7)
#define CPU_SUBTYPE_ARM_XSCALE          ((cpu_subtype_t) 8)
#define CPU_SUBTYPE_ARM_V7              ((cpu_subtype_t) 9)  /* ARMv7-A and ARMv7-R */
#define CPU_SUBTYPE_ARM_V7F             ((cpu_subtype_t) 10) /* Cortex A9 */
#define CPU_SUBTYPE_ARM_V7S             ((cpu_subtype_t) 11) /* Swift */
#define CPU_SUBTYPE_ARM_V7K             ((cpu_subtype_t) 12)
#define CPU_SUBTYPE_ARM_V8              ((cpu_subtype_t) 13)
#define CPU_SUBTYPE_ARM_V6M             ((cpu_subtype_t) 14) /* Not meant to be run under xnu */
#define CPU_SUBTYPE_ARM_V7M             ((cpu_subtype_t) 15) /* Not meant to be run under xnu */
#define CPU_SUBTYPE_ARM_V7EM            ((cpu_subtype_t) 16) /* Not meant to be run under xnu */
#define CPU_SUBTYPE_ARM_V8M             ((cpu_subtype_t) 17) /* Not meant to be run under xnu */
```